common/web/gesture-recognizer/src/inputEventEngine.ts

```diff
@@ -1,59 +1,52 @@
 /// <reference path="inputEventCoordinate.ts" />
+/// <reference path="gestureRecognizerConfiguration.ts" />
+/// <reference path="includes/events.ts" />
 
 namespace com.keyman.osk {
+  type Mutable<Type> = {
+    -readonly [Property in keyof Type]: Type[Property];
+  };
+
   export type InputHandler = (coord: InputEventCoordinate) => void;
 
-  export interface InputEventEngineConfig {
-    /**
-     * Specifies the element that input listeners should be attached to.
-     */
-    readonly eventRoot: HTMLElement;
-    /**
-     * Specifies the most specific common ancestor element of any event target
-     * that the InputEventEngine should consider.
-     */
-    readonly targetRoot: HTMLElement;
-
-    readonly coordConstrainedWithinInteractiveBounds: (coord: InputEventCoordinate) => boolean;
-
-    readonly inputStartHandler?:      InputHandler;
-    readonly inputMoveHandler?:       InputHandler;
-    readonly inputMoveCancelHandler?: InputHandler;
-    readonly inputEndHandler?:        InputHandler;
-  }
+  export abstract class InputEventEngine extends EventEmitter {
+    public static readonly INPUT_UPDATE_EVENT_NAME = "inputUpdate";
+
+    protected readonly config: GestureRecognizerConfiguration;
+
+    protected static preprocessConfig(config: GestureRecognizerConfiguration): GestureRecognizerConfiguration {
+      // Allows configuration pre-processing during this method.
+      let processingConfig: Mutable<GestureRecognizerConfiguration> = Object.assign({}, config);
+      processingConfig.mouseEventRoot = processingConfig.mouseEventRoot ?? processingConfig.targetRoot;
+      processingConfig.touchEventRoot = processingConfig.touchEventRoot ?? processingConfig.targetRoot;
+
+      return processingConfig;
+    }
```

Comment:
This is repeated in gestureRecognizer.ts. Why?

Reply:
For what it's worth, this is addressed in the followup (#6844) - it's the primary change in `inputEventEngine.ts` there.  The way this is done requires some of that code's other changes, though, so it's hard to move up.

So... what you see here was sort of a transient state while I was working out this aspect of the design.